### ISSUE TITLE
Making wetty a freebsd service

### ISSUE THREAD

**Issue by plato79** [butlerx/wetty#157]
I'm trying to make a freebsd init script for wetty. Here is my script:

```
#!/bin/sh
#
#

# PROVIDE: wetty
# REQUIRE: LOGIN
# KEYWORD: shutdown

# Add the following lines to /etc/rc.conf to enable wetty:
# wetty_enable="YES"

. /etc/rc.subr

name="wetty"
wetty=wetty_enable

load_rc_config $name

: ${wetty_enable="NO"}
: ${wetty_root:="/usr/local/wetty"}

pidfile="${wetty_root}/wetty.pid"
procname="/usr/local/bin/node"
command="/usr/sbin/daemon"
command_args="-f ${procname} ${wetty_root}/index.js"

command_args="${wetty_root}/index.js"

run_rc_command "$1"
```

The thing is I'm getting this error if I try to run it this way:

```
2019-01-13T11:31:35.734Z [Wetty] debug: key: undefined, cert: undefined, port: 3000, base: /wetty/
2019-01-13T11:31:35.740Z [Wetty] error: ENOENT: no such file or directory, stat '/dist/client/favicon.ico'
```

The problem is when I run it this way, it searches `favicon.ico` under the current directory's `dist/client` directory.

Though this line should tell rc script to execute this commands under this directory:

`: ${wetty_root:="/usr/local/wetty"}`

There is also another variable named `wetty_chdir` which provides same thing. But providing either of them doesn't work.

I know this problem is not directly related to wetty but it's about wetty installation under FreeBSD. So it'd be best if someone with scripting experience could help with this.

This problem doesn't exist when we install wetty as global so this could be fixed if #156 is fixed. If we execute `wetty` on the command line it finds its' own directory and there is no error at all.

**Comment by butlerx** [comment]
I think this may to longer be an issue of version `1.1.3` would you be able to confirm

**Comment by plato79** [comment]
Well, while finally wetty is working as standalone, there's still problem when calling it as a service...

It gives an error like this:

```
Starting wetty.
env: node: No such file or directory
/usr/local/etc/rc.d/wetty: WARNING: failed to start wetty
```

It seems it couldn't find node executable when starting as a service.. Really weird.

**Comment by butlerx** [comment]
Hmmm. Seems node isn't in your path.

**Comment by plato79** [comment]
But it is.. It is in /usr/local/bin and it's in the path:

`PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/games:/usr/local/sbin:/usr/local/bin`

**Comment by exislow** [comment]
Any solution?

**Comment by plato79** [comment]
TL;DR: It still needs a lot of work to make it a proper service.

Tried a different approach.. While it looks ok at first there is a bug.. Even when it's running as a service, after loading wetty page again closes the service.. I'm not sure what the problem is.

For example: run ssh, start the service, connect to the wetty, close ssh -> wetty stays active and it's still in the background when I check with `ps ax`.

But if you refresh the page or open a new wetty window, the service will be closed and any open wetty window will also close ( shows reconnect button ). 

I also didn't forward the output to anywhere so when you run the service, the messages log to the console, 

There is something going on with `env` though.. So just change the first line of `/usr/local/bin/wetty` like this:

`#! /usr/local/bin/node`


`/usr/local/etc/rc.d/wetty`:

```
#!/bin/sh
#
#

# PROVIDE: wetty
# REQUIRE: LOGIN
# KEYWORD: shutdown

# Add the following lines to /etc/rc.conf to enable wetty:
# wetty_enable="YES"

. /etc/rc.subr

name="wetty"
wetty=wetty_enable

load_rc_config $name

: ${wetty_enable="NO"}
: ${wetty_root:="/usr/local/wetty"}
: ${wetty_chdir:="/usr/local/wetty"}

pidfile="${wetty_root}/wetty.pid"
# procname="/usr/local/bin/node"
procname="/usr/local/bin/wetty -b /"
command="/usr/sbin/daemon"
# command_args="-f ${procname} -p ${pidfile}"


# command="/usr/local/bin/node"
command_args="/usr/local/bin/wetty -b /"

wetty_start() {
        /usr/sbin/daemon -p ${pidfile} ${command} ${command_args}
}

wetty_stop() {
        if [ -e "${pidfile}" ]; then
                kill -s TERM `cat ${pidfile}`
        else
                echo "${name} is not running"
        fi

}

wetty_status() {
        if [ -e "${pidfile}" ]; then
                echo "${name} is running as pid `cat ${pidfile}`"
        else
                echo "${name} is not running"
        fi
}

run_rc_command "$1"
```

**Comment by plato79** [comment]
Though, I said at first it doesn't seem to work, after using for a while it does work. Only problem I see is if you run the service via command line that command line needs to stay open. But if it's enabled at the startup, it doesn't close. I'm not sure what is the reason for this, because I'm executing under daemon.

There was one thing I missed though, from the wetty_start you need to remove ${command}.. It was from my tryouts. The final code is like this:

```
#!/bin/sh
#
#

# PROVIDE: wetty
# REQUIRE: LOGIN
# KEYWORD: shutdown

# Add the following lines to /etc/rc.conf to enable wetty:
# wetty_enable="YES"

. /etc/rc.subr

name="wetty"
wetty=wetty_enable

load_rc_config $name

: ${wetty_enable="NO"}
: ${wetty_root:="/usr/local/wetty"}
: ${wetty_chdir:="/usr/local/wetty"}

pidfile="${wetty_root}/wetty.pid"
# procname="/usr/local/bin/node"
procname="/usr/local/bin/wetty -b /"
command="/usr/sbin/daemon"
# command_args="-f ${procname} -p ${pidfile}"


# command="/usr/local/bin/node"
command_args="/usr/local/bin/wetty -b /"

wetty_start() {
        /usr/sbin/daemon -p ${pidfile} ${command_args}
}

wetty_stop() {
        if [ -e "${pidfile}" ]; then
                kill -s TERM `cat ${pidfile}`
        else
                echo "${name} is not running"
        fi

}

wetty_status() {
        if [ -e "${pidfile}" ]; then
                echo "${name} is running as pid `cat ${pidfile}`"
        else
                echo "${name} is not running"
        fi
}

run_rc_command "$1"
```
